overlord/ifacestate/handlers.go

```diff
@@ -1382,6 +1385,70 @@ func filterForSlot(slot *snap.SlotInfo) func(candSlots []*snap.SlotInfo) []*snap
 	}
 }
 
+// Keep this separate to allow mocking
+var isSnapVerified = func(st *state.State, snapID string) bool {
+	acc, err := assertstate.Publisher(st, snapID)
+	if err != nil {
+		logger.Debugf("cannot retrieve publisher assertion for %q: %v", snapID, err)
+		return false
+	}
+	return acc.Validation() == "verified"
+}
+
+func getAllowOptionAsString(inter string, tr *config.Transaction) (string, error) {
+	var value any
+	option := fmt.Sprintf("interface.%s.allow-auto-connection", inter)
+	err := tr.Get("core", option, &value)
+	if err != nil && !config.IsNoOption(err) {
+		return "", err
+	}
+
+	switch option := value.(type) {
+	case string:
+		return option, nil
+	case bool:
+		return strconv.FormatBool(option), nil
+	}
+	return "", nil
+}
+
+func isSnapSlotAllowed(st *state.State, snapID string, slot *snap.SlotInfo, tr *config.Transaction) (bool, error) {
```

Comment:
I still think that it would make sense to short circuit things here as well:

```
if slot.Interface != "x11" {
      return true, nil
}
```

Reply:
done!